src/Tasks/Microsoft.NET.Build.Extensions.Tasks/GetDependsOnNETStandard.net46.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if NET46
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.NET.Build.Tasks
+{
+    public partial class GetDependsOnNETStandard
+    {
+        private static Guid s_importerGuid = new Guid("7DAC8207-D3AE-4c75-9B67-92801A497D44");
+
+        // This method cross-compiles for desktop to avoid using System.Reflection.Metadata (SRM).
+        // We do this because we don't want the following:
+        //  - additional size of SRM and its closure
+        //  - load / JIT cost of SRM and its closure
+        //  - deal with bindingRedirects/unification needed to load SRM's closure.
+
+        internal static bool GetFileDependsOnNETStandard(string filePath)
+        {
+            // Ported from Microsoft.Build.Tasks.AssemblyInformation
+            if (Environment.OSVersion.Platform == PlatformID.Unix || Environment.OSVersion.Platform == PlatformID.MacOSX)
+            {
+                // on Unix/Mac (mono) use ReflectionOnlyLoadFrom to examine dependencies
+                // known issue: since this doesn't create an isolated app domain this will fail if more than one
+                // assembly with the same name is analyzed by this task.  The same issue exists in ResolveAssemblyReferences
+                // so we are not attempting to fix it here.
+                var assembly = Assembly.ReflectionOnlyLoadFrom(filePath);
+
+                foreach(var referencedAssembly in assembly.GetReferencedAssemblies())
+                {
+                    if (referencedAssembly.Name.Equals(NetStandardAssemblyName, StringComparison.Ordinal))
+                    {
+                        return true;
+                    }
+                }
+            }
+            else
+            {
+                // on Windows use CLR's unmanaged metadata API.
+                // Create the metadata dispenser and open scope on the source file.
+                IMetaDataDispenser metadataDispenser = (IMetaDataDispenser)new CorMetaDataDispenser();
+                IMetaDataAssemblyImport assemblyImport = (IMetaDataAssemblyImport)metadataDispenser.OpenScope(filePath, 0, ref s_importerGuid);
```

Comment:
Ok, this code was copied so I tried to make minimal changes.  I went through and sprinkled `var` where it was appropriate.